dotnet/ios/ios12/MarsHabitatPricePredictor/MarsHabitatPricePredictor/ViewController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using MarsHabitatPricePredictor.DataSources;
+
+namespace MarsHabitatPricePredictor;
+
+/// <summary>
+/// Main view controller for the MarsHabitatPricer app. Uses a `UIPickerView` to gather user inputs. The model's output is the predicted price.
+/// </summary>
+public partial class ViewController : UIViewController, IUIPickerViewDelegate
+{
+    private readonly MarsHabitatPricer model = new ();
+
+    // Data source for the picker.
+    private readonly PickerDataSource pickerDataSource = new ();
+
+    // Formatter for the output.
+    private readonly NSNumberFormatter priceFormatter = new ()
+    {
+        NumberStyle = NSNumberFormatterStyle.Currency,
+        MaximumFractionDigits = 0,
+        UsesGroupingSeparator = true,
+        Locale = new NSLocale ("en_US")
+    };
+
+    protected ViewController (IntPtr handle) : base (handle) { }
+
+    public override void ViewDidLoad ()
+    {
+        base.ViewDidLoad ();
+
+        this.pickerView.Delegate = this;
+        this.pickerView.DataSource = this.pickerDataSource;
+
+        // set default values
+        var features = Enum.GetValues (typeof (Feature)).Cast<Feature> ();
+        foreach (var feature in features)
+        {
+            this.pickerView.Select (2, (int)feature, false);
+        }
+
+        this.UpdatePredictedPrice ();
+    }
+
+    /// <summary>
+    /// The main logic for the app, performing the integration with Core ML.
+    /// First gather the values for input to the model. Then have the model generate
+    /// a prediction with those inputs. Finally, present the predicted value to the user.
+    /// </summary>
+    private void UpdatePredictedPrice ()
+    {
+        var solarPanels = this.pickerDataSource.Value (SelectedRow (Feature.SolarPanels), Feature.SolarPanels);
+        var greenhouses = this.pickerDataSource.Value (SelectedRow (Feature.Greenhouses), Feature.Greenhouses);
+        var size = this.pickerDataSource.Value (SelectedRow (Feature.Size), Feature.Size);
+
+        var marsHabitatPricerOutput = this.model.GetPrediction (solarPanels, greenhouses, size, out NSError? error);
+        if (error is not null)
+        {
+            throw new Exception ("Unexpected runtime error.");
```

Comment:
This is more helpful:

```suggestion 
            throw new Exception ($"Unexpected runtime error: {error}");
```